From 2e44b2c1f2504549329a0534e26cb274d326a54e Mon Sep 17 00:00:00 2001
From: Sandor Yu <Sandor.yu@nxp.com>
Date: Thu, 25 Jan 2018 16:27:54 +0800
Subject: [PATCH 3258/5242] MLK-17461-3: hdp: Remove pixel clock root setting

commit  aae9e35d919948fa87de72865cb0730d46e28e75 from
https://source.codeaurora.org/external/imx/linux-imx.git

HDMI pixel clock reparent function have implemented in dts,
remove clock root set function from hdp driver.

Signed-off-by: Sandor Yu <Sandor.yu@nxp.com>
Reviewed-by: Robby Cai <robby.cai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/hdp/imx-hdp.c |   14 --------------
 drivers/gpu/drm/imx/hdp/imx-hdp.h |    1 -
 2 files changed, 15 deletions(-)

diff --git a/drivers/gpu/drm/imx/hdp/imx-hdp.c b/drivers/gpu/drm/imx/hdp/imx-hdp.c
index fedd775..fde815a 100644
--- a/drivers/gpu/drm/imx/hdp/imx-hdp.c
+++ b/drivers/gpu/drm/imx/hdp/imx-hdp.c
@@ -248,18 +248,6 @@ void imx8qm_phy_reset(sc_ipc_t ipcHndl, u8 reset)
 		pr_err("SC_R_HDMI PHY reset failed %d!\n", sciErr);
 }
 
-void imx8qm_hdmi_set_clock_root(sc_ipc_t ipcHndl)
-{
-	/* set clock source from av pll */
-	/* those clock default source from dig pll */
-	/* HDMI DI Pixel Link Mux Clock  */
-	sc_pm_set_clock_parent(ipcHndl, SC_R_HDMI, SC_PM_CLK_MISC0, 2);
-	/* HDMI DI Pixel Link Clock  */
-	sc_pm_set_clock_parent(ipcHndl, SC_R_HDMI, SC_PM_CLK_MISC1, 2);
-	/* HDMI DI Pixel Clock  */
-	sc_pm_set_clock_parent(ipcHndl, SC_R_HDMI, SC_PM_CLK_MISC3, 2);
-}
-
 int imx8qm_clock_init(struct hdp_clks *clks)
 {
 	struct imx_hdp *hdp = clks_to_imx_hdp(clks);
@@ -932,7 +920,6 @@ static int imx8qm_hdp_swrite(struct hdp_mem *mem, unsigned int addr, unsigned in
 	.pixel_link_mux = imx8qm_pixel_link_mux,
 
 	.clock_init = imx8qm_clock_init,
-	.set_clock_root = imx8qm_hdmi_set_clock_root,
 	.ipg_clock_set_rate = imx8qm_ipg_clock_set_rate,
 	.ipg_clock_enable = imx8qm_ipg_clock_enable,
 	.ipg_clock_disable = imx8qm_ipg_clock_disable,
@@ -1090,7 +1077,6 @@ static int imx_hdp_imx_bind(struct device *dev, struct device *master,
 		DRM_ERROR("Failed to initialize clock %d\n", ret);
 		return ret;
 	}
-	imx_hdp_call(hdp, set_clock_root, hdp->ipcHndl);
 
 	ret = imx_hdp_call(hdp, clock_init, &hdp->clks);
 	if (ret < 0) {
diff --git a/drivers/gpu/drm/imx/hdp/imx-hdp.h b/drivers/gpu/drm/imx/hdp/imx-hdp.h
index 5ece813..1afd228 100644
--- a/drivers/gpu/drm/imx/hdp/imx-hdp.h
+++ b/drivers/gpu/drm/imx/hdp/imx-hdp.h
@@ -95,7 +95,6 @@ struct hdp_ops {
 	void (*pixel_link_mux)(state_struct *state, struct drm_display_mode *mode);
 
 	int (*clock_init)(struct hdp_clks *clks);
-	void (*set_clock_root)(sc_ipc_t ipcHndl);
 	int (*ipg_clock_enable)(struct hdp_clks *clks);
 	void (*ipg_clock_disable)(struct hdp_clks *clks);
 	void (*ipg_clock_set_rate)(struct hdp_clks *clks);
-- 
1.7.9.5

